### PR TITLE
addon_folders.ts: assign scheme to URIs created for mockups

### LINKED ISSUE #146: Addons are not found if their folders are not on the system drive.
If you populate `blender.addonFolders` with relative paths, like this:

```
"blender.addonFolders": [
	"blender_addons/megaddon",
	"blender_addons/blend1/blend1_addon",
	"blender_addons/botaniq/botaniq_addon",

        ...
        
	"blender_addons/memsaver/memsaver_addon",
],
```

This extension does not recognize the drives the addons are present in. It automatically defaults to the system drive. Addons will not be found and will not be loaded, like they don't exist.

### PR DESCRIPTION
**Pull request by kiraacorsac**
Correctly constructs URI objects that are passed to functions related to searching addon files.

Additionally, vscode dependency is updated to a non-depricated version, compatible with the same engine version, and typescript is updated to a version that is compatible with that version.

Small changes in addon_folder.ts are to make the new typescript version happy. 

Closes #146

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export function handleErrors(func: () => Promise<void>) {
         try {
             await func();
         }
-        catch (err) {
+        catch (err: any) {
```
Comment (Griperis) [inline src/utils.ts:49]: This seems like a drive-by typing change.

Comment (kiraacorsac) [reply src/utils.ts:49]: looking at MR now, it should have been two MRs with the a "assign a scheme" depending on the "update packages". I can fix it if needed - I was kinda in a hurry to compile a working version for myself.

**Comment by Griperis** [comment]
@kiraacorsac can you resurrect this, so this can get into a newer version? I think the module update might still be required. 

CC @JacquesLucke

**Comment by Mateusz-Grzelinski** [comment]
I have some time to refresh the issue, but I have no idea what it fixes and how to test it.  Any tips?

File: src/addon_folder.ts [hunk 2]
```diff
@@ -157,7 +156,7 @@ async function foldersToWorkspaceFoldersMockup(folders: string[]) {
 
         mockups.push({
             "name" : path.basename(absolutePath),
-            "uri": vscode.Uri.parse(absolutePath),
+            "uri": vscode.Uri.from({ scheme: "file", path: absolutePath }),
```
Comment (Griperis) [inline src/addon_folder.ts:159]: @Mateusz-Grzelinski I think this line is the most important - `scheme: "file"`, without it the URI doesn't work correctly when the addons are loaded from drive other than the extension. @kiraacorsac can confirm this, he had issues with it.

Comment (kiraacorsac) [reply src/addon_folder.ts:159]: Yes, that is the issue. Hopefully I remember this correctly, but `Uri.parse` does not behave correctly as it interprets a non-system drive letter as a protocol (`D:\path` becomes just `path` with `D` protocol. `Uri.from` makes sure the path is interpreted correctly.

**Comment by Mateusz-Grzelinski** [comment]
Finally I was able to reproduce the problem. Please next time say that is it about [`blender.addonFolders`](vscode://settings/blender.addonFolders) 
* Merged with newest master
* Updated Changelog
* I had to reinstall dependencies for development, no big deal
* Are we dropping support for older VS code versions by updating dependencies?
  * do I see correctly that oldest vs code version we support is now `"^1.28.0"`?
  * I have very little experience managing packages for typescript, I am depending on you to make `package.json` correct
* I got error, fixed in [cfddba7](https://github.com/JacquesLucke/blender_vscode/pull/147/commits/cfddba70294615778af417a6c3af7a31e418b095) - it was pretty much random google fix, anyone knows why it happens?
```
node_modules/@types/vscode/index.d.ts:19037:9 - error TS2583: Cannot find name 'AsyncIterable'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2018' or later.

19037   text: AsyncIterable<string>;
              ~~~~~~~~~~~~~

[14:15:26] Found 1 error. Watching for file changes.
```
* one time I got this error, what indicates completely different issue on hand:
```
RuntimeError: Can't listen for client connections: [WinError 10013] An attempt was made to access a socket in a way forbidden by its access permissions
```

**Please Review the changes!**

File: src/utils.ts (hunk 1 above)
Comment (Mateusz-Grzelinski) [reply src/utils.ts:49]: I kept the change, but I think the `any` is no longer necessary after merge

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,12 @@
 
 ## Unreleased
 
+### Changed
+- Updated dependencies. Now oldest supported VS Code version is `1.28.0`. ([#137](https://github.com/JacquesLucke/blender_vscode/pull/147))
+
+### Fixed
+- Path to addon indicated by [`blender.addonFolders`](vscode://settings/blender.addonFolders) now works corectly for non `C:` drive on Windows ([#137](https://github.com/JacquesLucke/blender_vscode/pull/147))
```
Comment (kiraacorsac) [inline CHANGELOG.md:9]: Non-system, not non `C:`. System drive can by any letter, it's just *usually* `C:.`.

**Comment by kiraacorsac** [comment]
> Finally I was able to reproduce the problem. Please next time say that is it about `blender.addonFolders` - I have never used it before.

It is both in the merge request title and the issue it closes, but yeah.

>     * I had to reinstall dependencies for development, no big deal

Yes, as `package.json` got updated this is needed.

>     * Are we dropping support for older VS code versions by updating dependencies?
>       
>       * do I see correctly that oldest vs code version we support is now `"^1.28.0"`?
Yes. 1.28 is 5 years old at this point, hopefully it's OK. The update is needed as it's the lowest version where `vscode.Uri` does what it's supposed to do.


>     * I got error, fixed in [cfddba7](https://github.com/JacquesLucke/blender_vscode/pull/147/commits/cfddba70294615778af417a6c3af7a31e418b095) - it was pretty much random google fix, anyone knows why it happens?

Not sure why it appears now, propably some typescript weirdness. Cleaner fix should be bumping the `es6` as a whole, by setting `lib: ['es2018']` or `lib: ['es2020']`, but I haven't tried it. That way you avoid having two libraries, one pulled only for a single type.

**Comment by kiraacorsac** [comment]
I will try to rebuild it later this week and let you know whether everything works as expected. I don't have the repo for this extension running right now.

**Comment by Mateusz-Grzelinski** [comment]
the vs code samples are even more aggressive when it comes to dependency version, so i think we do not need to worry https://github.com/microsoft/vscode-extension-samples/blob/main/helloworld-sample/package-lock.json

**Comment by Mateusz-Grzelinski** [comment]
I updated branch to the newest master yet again. 
@kiraacorsac after you make the change I will test the changes on linux and windows and then ask Jack to do a realese.

**Comment by Mateusz-Grzelinski** [comment]
@kiraacorsac recently i was developing using typescript 5.5.2 and es2020 - worked totaly fine so I commited this change here. 
I would like to merge the change and ask Jack for release. Can you confirm if it looks good?

**Comment by kiraacorsac** [comment]
Thank you for resurrecting this @Mateusz-Grzelinski, it works as expected when installed as VSIX on windows, Bledner 3.6 and Blender 4.1.

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,6 +14,7 @@
 - Remove broken links in addon and extension dir (#172)
 
 ### Changed 
+- Updated dependencies. Now oldest supported VS Code version is `1.28.0`. ([#137](https://github.com/JacquesLucke/blender_vscode/pull/147))
```
Comment (JacquesLucke) [inline CHANGELOG.md:17]: When changing the minimum supported version, it's typically good to mention when that version was released.

**Review by JacquesLucke — APPROVED** [review]
LGTM besides the one inline comment.
